### PR TITLE
[IMP] hr_holidays: improve stress days display

### PR DESCRIPTION
**Pull request by Bertrand2**
Before this commit stress days were displayed as colored rectangles on the calendar which made it ambiguous compared to leaves. After this commit the stress day have the number colored instead and the signification has been added to the legend on the side.

task-3226525

### CODE REVIEW

**Review by brboi — APPROVED** [review]
LGTM for `/web` part 

Please say a little word in the commit message about the changes on the date formatting ;-)

File: addons/hr_holidays/models/hr_employee.py [hunk 1]
```diff
@@ -281,6 +281,49 @@ def get_stress_days(self, start_date, end_date):
...
+            'startType': "datetime",
+            'title': sd.name,
+        }, stress_days))
+
+    @api.model
+    def get_bank_holidays_data(self, date_start, date_end):
+        self = self or self.env.user.employee_id
+        bank_holidays = self.env['resource.calendar.leaves'].search([
+            ('resource_id', '=', False),
+            ('company_id', 'in', self.company_id.ids + [False]),
```
Comment (sofiagvaladze) [inline addons/hr_holidays/models/hr_employee.py:304]:
```suggestion
            ('company_id', 'in', (self.company_id.id, False)),
```
same for `calendar_id`

File: addons/hr_holidays/static/src/views/calendar/year/calendar_year_popover.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import { CalendarYearPopover } from "@web/views/calendar/calendar_year/calendar_year_popover";
+
+export class TimeOffCalendarYearPopover extends CalendarYearPopover {}
+TimeOffCalendarYearPopover.components = { Dialog };
+TimeOffCalendarYearPopover.template = "web.CalendarYearPopover";
+TimeOffCalendarYearPopover.subTemplates = {
+    ...CalendarYearPopover.subTemplates,
+    body: "hr_holidays.StressDayCalendarYearPopover.body",
+    // stressDayRecord: "hr_holidays.StressDayCalendarYearPopover.stressDayRecord",
```
Comment (sofiagvaladze) [inline addons/hr_holidays/static/src/views/calendar/year/calendar_year_popover.js:13]: :angel:

**Review by sofiagvaladze — APPROVED** [review]
@Bertrand2 thanks for your work :raised_hands: 

Some minor comments/questions :innocent:

File: addons/hr_holidays/models/hr_employee.py [hunk 3]
```diff
@@ -281,6 +281,49 @@ def get_stress_days(self, start_date, end_date):
 
         return all_days
 
+    @api.model
+    def get_stress_days_data(self, date_start, date_end):
+        self = self or self.env.user.employee_id
```
[the remark this answers is not in the document]
Comment (Bertrand2) [reply addons/hr_holidays/models/hr_employee.py:286]: As discussed IRL, `self` is empty when the leave calendar is seen from the dashboard; self will only be used when going through an employee holidays smartbutton

File: addons/hr_holidays/models/hr_employee.py [hunk 4]
```diff
@@ -268,6 +268,14 @@ def action_time_off_dashboard(self):
             },
         }
 
+    def _get_contextual_employee(self):
+        if self.env.context.get('employee_id'):
+            return self.browse(self.env.context.get('employee_id'))
```
Comment (kba-odoo) [inline addons/hr_holidays/models/hr_employee.py:273]: As you know it does exists in the dict:
```suggestion
            return self.browse(self.env.context['employee_id'])
```

File: addons/hr_holidays/static/src/views/calendar/calendar_controller.js [hunk 5]
```diff
@@ -25,6 +25,12 @@ export class TimeOffCalendarController extends CalendarController {
         return this.model.employeeId;
     }
 
+    get filterPanelProps() {
+        const props = super.filterPanelProps;
+        props.employee_id = this.employeeId;
+        return props;
```
Comment (kba-odoo) [inline addons/hr_holidays/static/src/views/calendar/calendar_controller.js:31]:
```suggestion
        return {
            ...super.filterPanelProps,
            employee_id: this.employeeId,
        };
```

**Review by kba-odoo — CHANGES_REQUESTED** [review]
Some minor nitpicking :-)

However it doesn't seem to work the public holidays created as demo data:

image
image

Changing the starting hour to 8am instead of 9am shows it tho :thinking:

**Comment by Bertrand2** [comment]
coucou @kba-odoo , about the issue on the bank holiday not displaying, it's related to the task, though I indeed think there's some work to do to improve the behaviour of bank holidays on that point. I started a discussion about that issue on discord.

**Comment by kba-odoo** [comment]
robodoo r+

**Pull request closed without merging**